File: frontend/src/services/document.ts
import api from './api';

// 类型定义
export interface Document {
  id: number;
  filename: string;
  original_filename: string;
  file_type: string;
  course_id: number;
  subject: string | null;
  chapter_id: number | null;
  status: 'processing' | 'completed' | 'failed';
  error_message: string | null;
  chunk_count: number;
  created_at: string;
}

export interface KnowledgeChunk {
  id: number;
  content: string;
  chunk_index: number;
  subject: string | null;
  chapter_id: number | null;
}

export interface DocumentListResponse {
  documents: Document[];
  total: number;
}

// 文档 API
export const documentApi = {
  // 上传文档
  upload: (courseId: number, file: File) => {
    const formData = new FormData();
    formData.append('file', file);
    formData.append('course_id', String(courseId));
    return api.post<Document>('/document/upload', formData, {
      headers: {
        'Content-Type': 'multipart/form-data',
      },
    });
  },

  // 获取文档列表
  list: (courseId: number) =>
    api.get<DocumentListResponse>('/document/list', {
      params: { course_id: courseId },
    }),

  // 删除文档
  delete: (documentId: number) =>
    api.delete(`/document/${documentId}`),

  // 获取文档知识块
  getChunks: (documentId: number) =>
    api.get<KnowledgeChunk[]>(`/document/${documentId}/chunks`),
};

File: backend/app/schemas/document.py
"""文档相关的请求/响应模型"""

from datetime import datetime

from pydantic import BaseModel, Field

from backend.app.models.document import DocumentStatus, FileType


class DocumentUploadRequest(BaseModel):
    """文档上传请求（表单字段）"""

    course_id: int
    subject: str | None = None
    chapter_id: int | None = None


class DocumentResponse(BaseModel):
    """文档响应"""

    id: int
    filename: str
    original_filename: str
    file_type: FileType
    course_id: int
    subject: str | None
    chapter_id: int | None
    status: DocumentStatus
    error_message: str | None
    chunk_count: int
    created_at: datetime

    model_config = {"from_attributes": True}


class DocumentListResponse(BaseModel):
    """文档列表响应"""

    documents: list[DocumentResponse]
    total: int


class ChunkResponse(BaseModel):
    """知识块响应"""

    id: int
    content: str
    chunk_index: int
    subject: str | None
    chapter_id: int | None

    model_config = {"from_attributes": True}


class ChunkListResponse(BaseModel):
    """知识块列表响应"""

    chunks: list[ChunkResponse]
    total: int


class RetrievalRequest(BaseModel):
    """检索请求"""

    query: str = Field(..., min_length=1)
    course_id: int
    subject: str | None = None
    chapter_id: int | None = None
    top_k: int = Field(default=10, ge=1, le=50)
